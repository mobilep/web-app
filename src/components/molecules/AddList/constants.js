export const keyMap = {
	ENTER: 13,
	UP_ARROW: 38,
	DOWN_ARROW: 40,
	TAB: 9,
};

export default {
	keyMap,
};
