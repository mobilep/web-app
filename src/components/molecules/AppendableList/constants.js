export const LIST_TYPES = {
	CHECK: 'check',
	STEP: 'step',
};

export default {
	LIST_TYPES,
};
