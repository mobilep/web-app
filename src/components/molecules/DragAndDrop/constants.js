export const DRAG_START_EVENTS = ['dragover', 'dragenter', 'dragstart'];
export const DRAG_END_EVENTS = ['dragleave', 'dragend', 'drop'];
