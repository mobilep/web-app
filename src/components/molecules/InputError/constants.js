export const POSITIONS = {
	TOP_LEFT_POP_UP: 'topLeftPopUp',
	TOP_RIGHT_POP_UP: 'topRightPopUp',
	BOTTOM_RIGHT_POP_UP: 'bottomRightPopUp',
	BOTTOM_LEFT_POP_UP: 'bottomLeftPopUp',
	RIGHT_POP_UP: 'rightPopUp',
	LEFT_POP_UP: 'leftPopUp',
};

export const COLOR = {
	RED_ICON: 'red',
	WHITE_ICON: 'white',
};

export default {
	POSITIONS,
	COLOR,
};
