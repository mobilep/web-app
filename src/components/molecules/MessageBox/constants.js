export const keyMap = {
	ENTER: 'enter',
	BACKSPACE: 'backspace',
};

export default {
	keyMap,
};
