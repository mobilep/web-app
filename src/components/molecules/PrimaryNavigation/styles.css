.PrimaryNavigation {
	align-items: center;
	display: flex;
	background-color: var(--tomatoRed);
	z-index: 1;
	padding: 0 20px;
	flex-shrink: 0;
	height: var(--primary-navigation-height-desktop);
}

.PrimaryNavigation-logo {
	margin-right: 20px;
}

.PrimaryNavigation-logo svg {
	display: block;
	width: 40px;
	height: 40px;
	margin: auto;
}

.PrimaryNavigation-nav {
	display: flex;
	text-align: center;
}

.PrimaryNavigation-nav a {
	margin-right: 10px;
}

.PrimaryNavigation-logOut {
	margin-left: auto;
}

@media (max-width: 768px) {
	.PrimaryNavigation {
		background-color: var(--white);
		border-top: 1px solid var(--grey2);
		padding: 0;
	}

	.PrimaryNavigation-nav {
		width: 100%;
		justify-content: space-evenly;
	}

	.PrimaryNavigation-nav > a {
		flex: 0 0 25%;
		margin: 0;
	}

	.PrimaryNavigation-logo {
		display: none;
	}

	.PrimaryNavigation-logOut {
		display: none;
	}

	.PrimaryNavigationItem:nth-child(5) {
		display: none;
	}
}
