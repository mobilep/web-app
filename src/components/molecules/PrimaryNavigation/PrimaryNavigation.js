import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { DialogWithButtonGroup, PrimaryNavigationItem, IconWithCounter } from '../..';
import {
	LogoIcon,
	LogOutIcon,
	ScenariousIcon,
	TeamIcon,
	ProfileIcon,
	SettingsIcon,
	InboxIcon,
} from '../../../icons';
import constants from '../../../constants';

import './styles.css';

const {
	INBOX,
	SCENARIOS,
	TEAM,
	PROFILE,
	SETTINGS,
	LEARNER_REPORT,
} = constants.routes;

export default class PrimaryNavigation extends Component {

	dialogRef = React.createRef();

	getCurrentDialogRef = () => this.dialogRef.current;
	getClassName = (props) => classNames('PrimaryNavigation', props.className)

	handleDialogCancelButtonClick = () => this.getCurrentDialogRef().close();
	handleDialogConfirmButtonClick = () => this.props.onSignOut();
	handleLogOutButtonClick = () => this.getCurrentDialogRef().showModal();

	render () {
		const { common, primaryNav } = this.props.content;

		return (
			<div className={this.getClassName(this.props)}>

				<div className='PrimaryNavigation-logo'>
					<LogoIcon />
				</div>

				<nav className='PrimaryNavigation-nav'>
					<PrimaryNavigationItem to={SCENARIOS}>
						<IconWithCounter icon={<ScenariousIcon />} counter={this.props.unreadScenariosCount} />
						<span>{primaryNav.scenarious}</span>
					</PrimaryNavigationItem>

					<PrimaryNavigationItem to={INBOX}>
						<IconWithCounter icon={<InboxIcon />} counter={this.props.unreadInboxCount} />
						<span>{primaryNav.inbox}</span>
					</PrimaryNavigationItem>

					<PrimaryNavigationItem to={TEAM}>
						<TeamIcon />
						<span>{primaryNav.team}</span>
					</PrimaryNavigationItem>

					<PrimaryNavigationItem
						to={LEARNER_REPORT}
						isActive={(match, location) => location.pathname.startsWith(PROFILE)}
					>
						<ProfileIcon />
						<span>{primaryNav.profile}</span>
					</PrimaryNavigationItem>

					<PrimaryNavigationItem to={SETTINGS}>
						<SettingsIcon />
						<span>{primaryNav.settings}</span>
					</PrimaryNavigationItem>
				</nav>

				<div className='PrimaryNavigation-logOut'>
					<PrimaryNavigationItem onClick={this.handleLogOutButtonClick}>
						<LogOutIcon />
						<span>{primaryNav.logOut}</span>
					</PrimaryNavigationItem>
				</div>

				<DialogWithButtonGroup
					actionButtonLabel={common.logOutButton}
					cancelButtonLabel={common.cancel}
					onActionButtonClick={this.handleDialogConfirmButtonClick}
					onCancelButtonClick={this.handleDialogCancelButtonClick}
					ref={this.dialogRef}
				>
					{common.logOutMessage}
				</DialogWithButtonGroup>

			</div>
		);
	}
}

PrimaryNavigation.propTypes = {
	className: PropTypes.string,
	content: PropTypes.object,
	onSignOut: PropTypes.func.isRequired,
	unreadInboxCount: PropTypes.number,
	unreadScenariosCount: PropTypes.number,
};
