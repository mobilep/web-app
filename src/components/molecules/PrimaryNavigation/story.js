import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { number } from '@storybook/addon-knobs';
import StoryRouter from 'storybook-react-router';

import PrimaryNavigation from './PrimaryNavigation';
import texts from '../../../constants/content';

const { common, primaryNav } = texts.en;

storiesOf('2. Molecules / PrimaryNavigation', module)
	.addDecorator(StoryRouter())
	.add('default', () => (
		<div style={{ width: '64px', height: '850px' }}>
			<PrimaryNavigation
				content={{ common, primaryNav }}
				onSignOut={action('Sign out click')}
				unreadInboxCount={number('Unread count', 4)}
			/>
		</div>
	));
