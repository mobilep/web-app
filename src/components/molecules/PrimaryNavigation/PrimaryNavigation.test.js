import React from 'react';
import { shallow } from 'enzyme';

import PrimaryNavigation from './PrimaryNavigation';
import constants from '../../../constants/content';

const { common, primaryNav } = constants.en;

describe('Molecule - PrimaryNavigation', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<PrimaryNavigation
				content={{ common, primaryNav }}
				onSignOut={() => {}}
				unreadInboxCount={4}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
