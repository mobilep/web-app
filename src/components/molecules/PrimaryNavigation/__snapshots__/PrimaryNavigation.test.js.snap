// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - PrimaryNavigation Should render 1`] = `
<div
  className="PrimaryNavigation"
>
  <div
    className="PrimaryNavigation-logo"
  >
    <LogoIcon />
  </div>
  <nav
    className="PrimaryNavigation-nav"
  >
    <PrimaryNavigationItem
      to="/scenarios"
    >
      <IconWithCounter
        icon={<ScenariousIcon />}
      />
      <span>
        SCENARIOS
      </span>
    </PrimaryNavigationItem>
    <PrimaryNavigationItem
      to="/inbox"
    >
      <IconWithCounter
        counter={4}
        icon={<InboxIcon />}
      />
      <span>
        MESSAGES
      </span>
    </PrimaryNavigationItem>
    <PrimaryNavigationItem
      to="/groups"
    >
      <TeamIcon />
      <span>
        TEAMS
      </span>
    </PrimaryNavigationItem>
    <PrimaryNavigationItem
      isActive={[Function]}
      to="/profile/learner-report"
    >
      <ProfileIcon />
      <span>
        PROFILE
      </span>
    </PrimaryNavigationItem>
    <PrimaryNavigationItem
      to="/settings"
    >
      <SettingsIcon />
      <span>
        SETTINGS
      </span>
    </PrimaryNavigationItem>
  </nav>
  <div
    className="PrimaryNavigation-logOut"
  >
    <PrimaryNavigationItem
      onClick={[Function]}
    >
      <LogOutIcon />
      <span>
        SIGN OUT
      </span>
    </PrimaryNavigationItem>
  </div>
  <DialogWithButtonGroup
    actionButtonLabel="Sign out"
    cancelButtonLabel="Cancel"
    onActionButtonClick={[Function]}
    onCancelButtonClick={[Function]}
  >
    Are you sure you want to sign out?
  </DialogWithButtonGroup>
</div>
`;
